tensorflow_datasets/scripts/documentation/dataset_markdown_builder.py

```diff
@@ -0,0 +1,386 @@
+  return None  # Fuse the sections together if no configs are available
+
+
+# --------------------------- Single builder ---------------------------
+
+def display_builder(builder, sections):
+  return ("\n").join([textwrap.dedent(section.make(builder))
+                      for section in sections if section.make(builder)])
+
+
+# --------------------------- Builder configs ---------------------------
+
+def display_all_builders(nightly_doc_util, builders, all_sections):
+  # For each fields, extract if the field is shared or unique accross builder.
+  common_sections = []
+  unique_sections = []
+  for section in all_sections:
+    if len(set(section.get_signature(b) for b in builders)) == 1:
+      common_sections.append(section)
+    else:
+      unique_sections.append(section)
+
+  common_builder_str = display_builder(next(iter(builders)), common_sections)
+
+  unique_builder_str = []
+  for i, builder in enumerate(builders):
+    header_suffix = " (default config)" if i == 0 else ""
+    nightly_str = (" " + nightly_doc_util.icon) \
+        if nightly_doc_util.is_config_nightly(builder) else ""
+    unique_builder_str.append(
+        f"## {builder.name}/{builder.builder_config.name}"
+        f"{header_suffix}{nightly_str}\n")
+    unique_builder_str.append(display_builder(builder, unique_sections))
+  unique_builder_str = ("\n").join(unique_builder_str)
+
+  return common_builder_str + "\n" + unique_builder_str
+
+
+# --------------------------- Main page ---------------------------
+
+def display_builder_configs(
+    builder,
+    nightly_doc_util,
+    config_builders,
+    all_sections
+):
+  # First case: Single builder
+  if not builder.builder_config:
+    return display_builder(builder, all_sections)
+  # Second case: Builder configs
+  return display_all_builders(nightly_doc_util, config_builders, all_sections)
+
+
+def display_nightly_str(nightly_doc_util, builder):
+  if nightly_doc_util.is_builder_nightly(builder):
+    return f"""\
+      Note: This dataset was added recently and is only available in our
+      `tfds-nightly` package  {nightly_doc_util.icon}.
+      """
+  if nightly_doc_util.has_nightly(builder):
+    return f"""\
+      Note: This dataset has been updated since the last stable release.
+      The new versions and config marked with {nightly_doc_util.icon}
+      are only available in the `tfds-nightly` package.
+      """
+  return ""
+
+
+def display_manual_instructions(builder):
+  if builder.MANUAL_DOWNLOAD_INSTRUCTIONS:
+    return "Warning: Manual download required. See instructions below.\n"
+  return ""
+
+
+def display_dataset_heading(builder):
+  return textwrap.dedent(
+      f"""
+      # `{builder.name}`
+      """
+  )
+
+
+def get_markdown_string(
+    builder: tfds.core.DatasetBuilder,
+    config_builders: List[tfds.core.DatasetBuilder],
+    visu_doc_util,
+    nightly_doc_util,
+) -> str:
+
+  all_sections = [
+      Section(get_description, display_description),
```

Comment:
If you feel motivated, I think it would make sense to refactor this as subclass, to have:

```py
class HomepageSection(Section):

  def get_key(self, builder):
    return builder.info.homepage

  def display(self, builder):
    return f"""\
      *   **Homepage**: [{builder.info.homepage}]({builder.info.homepage})
    """
```

And here:

```py
all_sections: List[Section] = [
    ConfigDescriptionSection(),
    HomepageSection(),
    VersionSection(nightly_doc_util),
    ...
]
```

Previously it wasn't possible as mako do not support class.

I should have catched this sooner so no worries if you don't want to do it. I can try when merging this internally.

Reply:
Done